File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-6.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int max = 0;
        int l = 0;
        int r = 0;
        HashSet<Character> charSet = new HashSet<>();
        while(r < s.length())
        {
            while(charSet.contains(s.charAt(r)))
            {
                charSet.remove(s.charAt(l));
                l += 1;
            }
            max = Math.max(max, r - l + 1);
            charSet.add(s.charAt(r));
            r += 1;
        }
        return max;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-6.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        d = {}
        longest= 0 
        for x in nums:
            if x-1 not in d.keys():
                len = 1
                while len+x in nums:
                    len = len + 1
                longest = max(longest, len)
        return longest
                